api/saml/saml.go

```diff
@@ -57,23 +82,34 @@ func (service *Service) ValidateAuthenticationResponse(encoded string) (string,
 	response, err := saml.ParseEncodedResponse(encoded)
 	if err != nil {
 		service.Log.WarnError(api.SamlParseError, err, api.LogFields{})
-		return "", err
+		return "", "", err
 	}
 
 	// err = response.Validate(&service.provider)
 	err = service.validate(response, string(authnResponseXML))
 	if err != nil {
 		service.Log.WarnError(api.SamlInvalid, err, api.LogFields{})
-		return "", err
+		return "", "", err
 	}
 
 	username := cleanName(response.Assertion.Subject.NameID.Value)
 	if username == "" {
 		service.Log.WarnError(api.SamlIdentifierMissing, err, api.LogFields{})
-		return "", err
+		return "", "", err
+	}
+
+	var sessionIndex string
+	for _, authnStatement := range response.Assertion.AuthnStatements {
+		if authnStatement.SessionIndex != "" {
+			sessionIndex = authnStatement.SessionIndex
+		}
+	}
+
+	if sessionIndex == "" {
+		service.Log.Warn("SAML Auth Response does not include a SessionIndex. WSO2 is probably misconfigured and SLO will likely not work correctly.", api.LogFields{})
```

Comment:
Also: we need to do some type of input validation here on `sessionIndex`, I suspect the underlying SAML library is not doing it. Should restrict sessionIndex to whatever the SAML spec restricts it to.

Reply:
Since it's possible to have WSO2 configured without SLO enabled, this is going to move to Info. There's no validation necessary for this, the spec allows any string for a sessionIndex.